pkg/render/node.go

```diff
@@ -398,6 +395,15 @@ func (c *nodeComponent) nodeDaemonset() *apps.DaemonSet {
 	annotations[typhaCAHashAnnotation] = AnnotationHash(c.typhaNodeTLS.CAConfigMap.Data)
 	annotations[nodeCertHashAnnotation] = AnnotationHash(c.typhaNodeTLS.NodeSecret.Data)
 
+	if c.cr.Spec.NodeMetricsPort != nil {
+		scrapingAnnotation := map[string]string{
+			"prometheus.io/scrape": "true",
+			"prometheus.io/port":   string(*c.cr.Spec.NodeMetricsPort),
+		}
+
+		annotations[prometheusHashAnnotation] = AnnotationHash(scrapingAnnotation)
```

Comment:
I don't think we need the annotation hash entry here for the scrape port. If the port changes, we're already going to be modifying an environment variable and the other annotation, meaning we'll trigger a rolling update without the need for the special hash annotation.

Reply:
Ok